Unable to compile template with MinGW

### Godot version

4.0.dev 4c3f7d1

### System information

Windows 10 21H1

### Issue description

Unable to compile `debug` and `release_debug` template with MinGW on Windows 10 because of the following error:
```
b'In file included from ./core/templates/hash_map.h:34,
                 from ./core/object/object.h:38,

    from ./core/variant/binder_common.h:35,
                 from ./core/object/method_bind.h:34,
                 from ./core/object/class_db.h:34,
                 from ./core/object/ref_counted.h:34,
                 from ./servers/text_server.h:34,
                 from modules\\text_server_adv\\text_server_adv.h:39,
                 from modules\\text_server_adv\\text_server_adv.cpp:31:
modules\\text_server_adv\\text_server_adv.cpp: In member function \'bool TextServerAdvanced::_ensure_cache_for_size(TextServerAdvanced::FontDataAdvanced*, const Vector2i&) const\':
./core/string/ustring.h:526:22: error: invalid cast to function type \'String()\'
  526 | #define TTR(m_value) (String())
./core/error/error_macros.h:94:26: note: in definition of macro \'DEBUG_STR\'
   94 | #define DEBUG_STR(m_msg) m_msg
      |                          ^~~~~
modules\\text_server_adv\\text_server_adv.cpp:1234:4: note: in expansion of macro \'ERR_FAIL_COND_V_MSG\'
 1234 |    ERR_FAIL_COND_V_MSG(error != 0, false, TTR("FreeType: Error initializing library:") + " \'" + String(FT_Error_String(error)) + "\'.");
      |    ^~~~~~~~~~~~~~~~~~~
modules\\text_server_adv\\text_server_adv.cpp:1234:43: note: in expansion of macro \'TTR\'
 1234 |    ERR_FAIL_COND_V_MSG(error != 0, false, TTR("FreeType: Error initializing library:") + " \'" + String(FT_Error_String(error)) + "\'.");
      |
  ^~~
./core/string/ustring.h:526:22: error: invalid cast to function type \'String()\'
  526 | #define TTR(m_value) (String())
./core/error/error_macros.h:94:26: note: in definition of macro \'DEBUG_STR\'
   94 | #define DEBUG_STR(m_msg) m_msg
      |                          ^~~~~
modules\\text_server_adv\\text_server_adv.cpp:1252:4: note: in expansion of macro \'ERR_FAIL_V_MSG\'
 1252 |    ERR_FAIL_V_MSG(false, TTR("FreeType: Error loading font:") + " \'" + String(FT_Error_String(error)) + "\'.");
      |    ^~~~~~~~~~~~~~
modules\\text_server_adv\\text_server_adv.cpp:1252:26: note: in expansion of macro \'TTR\'
 1252 |    ERR_FAIL_V_MSG(false, TTR("FreeType: Error loading font:") + " \'" + String(FT_Error_String(error)) + "\'.");
      |                          ^~~
'
```
This error introduced with 4c3f7d1

### Steps to reproduce

Checkout 4c3f7d1 and `scons platform=windows target=debug use_mingw=yes tools=no`

### Minimal reproduction project

_No response_

--- a/modules/text_server_adv/text_server_adv.cpp
+++ b/modules/text_server_adv/text_server_adv.cpp
@@ -1231,7 +1231,7 @@ _FORCE_INLINE_ bool TextServerAdvanced::_ensure_cache_for_size(FontDataAdvanced 
 		int error = 0;
 		if (!library) {
 			error = FT_Init_FreeType(&library);
-			ERR_FAIL_COND_V_MSG(error != 0, false, TTR("FreeType: Error initializing library:") + " '" + String(FT_Error_String(error)) + "'.");
+			ERR_FAIL_COND_V_MSG(error != 0, false, RTR("FreeType: Error initializing library:") + " '" + String(FT_Error_String(error)) + "'.");
 		}
 
 		memset(&fd->stream, 0, sizeof(FT_StreamRec));
@@ -1249,13 +1249,7 @@ _FORCE_INLINE_ bool TextServerAdvanced::_ensure_cache_for_size(FontDataAdvanced 
 		if (error) {
 			FT_Done_Face(fd->face);
 			fd->face = nullptr;
-			ERR_FAIL_V_MSG(false, TTR("FreeType: Error loading font:") + " '" + String(FT_Error_String(error)) + "'.");
-		}
-		fd->hb_handle = hb_ft_font_create(fd->face, nullptr);
-		if (fd->hb_handle == nullptr) {
-			FT_Done_Face(fd->face);
-			fd->face = nullptr;
-			ERR_FAIL_V_MSG(false, TTR("HarfBuzz: Error creating FreeType font object."));
+			ERR_FAIL_V_MSG(false, RTR("FreeType: Error loading font:") + " '" + String(FT_Error_String(error)) + "'.");
 		}
 
 		if (p_font_data->msdf) {
@@ -1283,6 +1277,8 @@ _FORCE_INLINE_ bool TextServerAdvanced::_ensure_cache_for_size(FontDataAdvanced 
 		} else {
 			FT_Set_Pixel_Sizes(fd->face, 0, fd->size.x * fd->oversampling);
 		}
+
+		fd->hb_handle = hb_ft_font_create(fd->face, nullptr);
 
 		fd->ascent = (fd->face->size->metrics.ascender / 64.0) / fd->oversampling * fd->scale;
 		fd->descent = (-fd->face->size->metrics.descender / 64.0) / fd->oversampling * fd->scale;
@@ -1598,14 +1594,11 @@ _FORCE_INLINE_ bool TextServerAdvanced::_ensure_cache_for_size(FontDataAdvanced 
 			FT_Done_MM_Var(library, amaster);
 		}
 #else
-		ERR_FAIL_V_MSG(false, TTR("FreeType: Can't load dynamic font, engine is compiled without FreeType support!");
+		ERR_FAIL_V_MSG(false, RTR("FreeType: Can't load dynamic font, engine is compiled without FreeType support!");
 #endif
 	} else {
 		// Init bitmap font.
 		fd->hb_handle = hb_bmp_font_create(fd, nullptr);
-		if (!fd->hb_handle) {
-			ERR_FAIL_V_MSG(false, TTR("HarfBuzz: Error creating bitmap font object."));
-		}
 	}
 	p_font_data->cache[p_size] = fd;
 	return true;

--- a/modules/text_server_fb/text_server_fb.cpp
+++ b/modules/text_server_fb/text_server_fb.cpp
@@ -686,7 +686,7 @@ _FORCE_INLINE_ bool TextServerFallback::_ensure_cache_for_size(FontDataFallback 
 		int error = 0;
 		if (!library) {
 			error = FT_Init_FreeType(&library);
-			ERR_FAIL_COND_V_MSG(error != 0, false, TTR("FreeType: Error initializing library:") + " '" + String(FT_Error_String(error)) + "'.");
+			ERR_FAIL_COND_V_MSG(error != 0, false, RTR("FreeType: Error initializing library:") + " '" + String(FT_Error_String(error)) + "'.");
 		}
 
 		memset(&fd->stream, 0, sizeof(FT_StreamRec));
@@ -704,7 +704,7 @@ _FORCE_INLINE_ bool TextServerFallback::_ensure_cache_for_size(FontDataFallback 
 		if (error) {
 			FT_Done_Face(fd->face);
 			fd->face = nullptr;
-			ERR_FAIL_V_MSG(false, TTR("FreeType: Error loading font:") + " '" + String(FT_Error_String(error)) + "'.");
+			ERR_FAIL_V_MSG(false, RTR("FreeType: Error loading font:") + " '" + String(FT_Error_String(error)) + "'.");
 		}
 
 		if (p_font_data->msdf) {
@@ -784,7 +784,7 @@ _FORCE_INLINE_ bool TextServerFallback::_ensure_cache_for_size(FontDataFallback 
 			FT_Done_MM_Var(library, amaster);
 		}
 #else
-		ERR_FAIL_V_MSG(false, TTR("FreeType: Can't load dynamic font, engine is compiled without FreeType support!");
+		ERR_FAIL_V_MSG(false, RTR("FreeType: Can't load dynamic font, engine is compiled without FreeType support!");
 #endif
 	}
 	p_font_data->cache[p_size] = fd;

